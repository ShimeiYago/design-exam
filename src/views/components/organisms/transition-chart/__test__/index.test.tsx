import { shallow, ShallowWrapper } from 'enzyme';
import { TransitionChart, Props } from '..';

let wrapper: ShallowWrapper<Props>;

const props: Props = {
  data: [
    { year: 2000, value: 17 },
    { year: 2001, value: 16.8 },
    { year: 2002, value: 16.8 },
    { year: 2003, value: 16.3 },
    { year: 2004, value: 15.8 },
    { year: 2005, value: 15.7 },
    { year: 2006, value: 15.6 },
    { year: 2007, value: 16.1 },
    { year: 2008, value: 15.5 },
    { year: 2009, value: 15.8 },
    { year: 2010, value: 15.8 },
    { year: 2011, value: 15.4 },
    { year: 2012, value: 15 },
    { year: 2013, value: 15.1 },
    { year: 2014, value: 14.9 },
    { year: 2015, value: 15.1 },
    { year: 2016, value: 15.2 },
    { year: 2017, value: 15.6 },
    { year: 2018, value: 15.5 },
    { year: 2019, value: 15.3 },
    { year: 2020, value: 15.2 },
    { year: 2021, value: 14.6 },
  ],
  loading: false,
  failed: false,
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    wrapper = shallow(<TransitionChart {...props} />);
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('loading case', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('basic', () => {
    wrapper.setProps({ failed: true });
    expect(wrapper).toMatchSnapshot();
  });
});
