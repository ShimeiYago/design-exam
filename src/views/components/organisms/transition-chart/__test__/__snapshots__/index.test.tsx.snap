// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<Spacing
  paddingHorizontal="15px"
  paddingVertical="20px"
>
  <TextWrapper
    bold={true}
    size="small"
  >
    進学者数の推移
  </TextWrapper>
  <ForwardRef
    height={500}
    width="100%"
  >
    <LineChart
      barCategoryGap="10%"
      barGap={4}
      data={
        Array [
          Object {
            "value": 17,
            "year": 2000,
          },
          Object {
            "value": 16.8,
            "year": 2001,
          },
          Object {
            "value": 16.8,
            "year": 2002,
          },
          Object {
            "value": 16.3,
            "year": 2003,
          },
          Object {
            "value": 15.8,
            "year": 2004,
          },
          Object {
            "value": 15.7,
            "year": 2005,
          },
          Object {
            "value": 15.6,
            "year": 2006,
          },
          Object {
            "value": 16.1,
            "year": 2007,
          },
          Object {
            "value": 15.5,
            "year": 2008,
          },
          Object {
            "value": 15.8,
            "year": 2009,
          },
          Object {
            "value": 15.8,
            "year": 2010,
          },
          Object {
            "value": 15.4,
            "year": 2011,
          },
          Object {
            "value": 15,
            "year": 2012,
          },
          Object {
            "value": 15.1,
            "year": 2013,
          },
          Object {
            "value": 14.9,
            "year": 2014,
          },
          Object {
            "value": 15.1,
            "year": 2015,
          },
          Object {
            "value": 15.2,
            "year": 2016,
          },
          Object {
            "value": 15.6,
            "year": 2017,
          },
          Object {
            "value": 15.5,
            "year": 2018,
          },
          Object {
            "value": 15.3,
            "year": 2019,
          },
          Object {
            "value": 15.2,
            "year": 2020,
          },
          Object {
            "value": 14.6,
            "year": 2021,
          },
        ]
      }
      layout="horizontal"
      margin={
        Object {
          "bottom": 5,
          "left": 20,
          "right": 30,
          "top": 20,
        }
      }
      reverseStackOrder={false}
      stackOffset="none"
      syncMethod="index"
    >
      <CartesianGrid
        strokeDasharray="3 3"
      />
      <XAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        dataKey="year"
        height={30}
        hide={false}
        label={
          Object {
            "offset": 0,
            "position": "insideBottomRight",
            "value": "年",
          }
        }
        mirror={false}
        orientation="bottom"
        padding={
          Object {
            "left": 0,
            "right": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="category"
        width={0}
        xAxisId={0}
      />
      <YAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        domain={
          Array [
            "dataMin",
            "dataMax",
          ]
        }
        height={0}
        hide={false}
        label={
          Object {
            "angle": -90,
            "position": "insideLeft",
            "value": "人数 (万人)",
          }
        }
        mirror={false}
        orientation="left"
        padding={
          Object {
            "bottom": 0,
            "top": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="number"
        width={60}
        yAxisId={0}
      />
      <Tooltip
        accessibilityLayer={false}
        allowEscapeViewBox={
          Object {
            "x": false,
            "y": false,
          }
        }
        animationDuration={400}
        animationEasing="ease"
        contentStyle={Object {}}
        coordinate={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        cursor={true}
        cursorStyle={Object {}}
        filterNull={true}
        isAnimationActive={true}
        itemStyle={Object {}}
        labelStyle={Object {}}
        offset={10}
        reverseDirection={
          Object {
            "x": false,
            "y": false,
          }
        }
        separator=" : "
        trigger="hover"
        useTranslate3d={false}
        viewBox={
          Object {
            "height": 0,
            "width": 0,
            "x": 0,
            "y": 0,
          }
        }
        wrapperStyle={Object {}}
      />
      <Line
        activeDot={
          Object {
            "r": 8,
          }
        }
        animateNewValues={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={false}
        dataKey="value"
        dot={true}
        fill="#fff"
        hide={false}
        isAnimationActive={true}
        label={false}
        legendType="line"
        points={Array []}
        stroke="#8884d8"
        strokeWidth={1}
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
    </LineChart>
  </ForwardRef>
</Spacing>
`;

exports[`Shallow Snapshot Tests basic 2`] = `
<Spacing
  paddingHorizontal="15px"
  paddingVertical="20px"
>
  <TextWrapper
    align="center"
    size="medium"
  >
    データの取得に失敗しました。しばらくしてから再度お試しください。
  </TextWrapper>
</Spacing>
`;

exports[`Shallow Snapshot Tests loading case 1`] = `
<Spacing
  paddingHorizontal="15px"
  paddingVertical="20px"
>
  <TextWrapper
    align="center"
    size="medium"
  >
    Loading...
  </TextWrapper>
</Spacing>
`;
